plugins/filter_aws/aws.c

```diff
@@ -102,6 +104,18 @@ static void expose_aws_meta(struct flb_filter_aws *ctx)
                     "aws." FLB_FILTER_AWS_HOSTNAME_KEY,
                     ctx->hostname);
     }
+
+    for (i = 0; i < ctx->tags_count; i++) {
+        /* len("aws.ec2.tags.") = 13 */
+        env_tag_name = flb_sds_create_size(13 + ctx->tag_keys_len[i]);
```

Comment:
optional nit: flb_sds_printf will always alloc at least 64 bytes if you look at the code... so for perf optimization you can only alloc this many bytes if its over 64, and then otherwise alloc 64.

Reply:
My take would be to not increase complexity because of performance in this case. This code should be called only once per instance tag at the initialization phase, so most likely < 10 times in the single fluent-bit execution lifetime.

Let me know what you think.

Reply:
Makes sense 👍